src/mpp/Dec.hpp

```diff
@@ -1172,6 +1172,27 @@ bool jump_read_key(BUF& buf, T... t)
 	return jump_find_key<PAIRS, FAMILY, PATH>(val, IS, buf, t...);
 }
 
+template<compact::Family FAMILY, size_t SUBRULE,
+	 class DST, class BUF, class = void>
+struct is_optional_or_variant_readable_in_place : std::false_type {};
+
+template<compact::Family FAMILY, size_t SUBRULE,
+	 class DST, class BUF>
+struct is_optional_or_variant_readable_in_place
+	<FAMILY, SUBRULE, DST, BUF,
+	 typename std::enable_if_t<(FAMILY == compact::MP_BOOL ||
+				    FAMILY == compact::MP_INT ||
+				    FAMILY == compact::MP_FLT) &&
```

Comment:
I think it should be expressed with Rule properties, not enumeration.
It seems that families that meet the condition `!FAMILY::has_ext && !FAMILY::has_data && !FAMILY::has_children)` can be read by value. Probably, it should be a new field at `BaseRule`.

And why didn't you included `MP_NIL`?

Reply:
Fixed.

I thought it would always get dispatched to other short paths.